lib/zypper.rb

```diff
@@ -27,32 +27,47 @@ class Zypper
 
   class <<self
     def isolated(options = {}, &block)
-      zypper_base = Dir.mktmpdir("machinery_zypper")
+      @zypp_base = Dir.mktmpdir("machinery_zypper")
       zypper = Zypper.new
 
       zypper.zypper_options = [
         "--non-interactive",
         "--no-gpg-checks",
-        "--root", zypper_base
+        "--root", @zypp_base
       ]
 
       if options[:arch]
-        zypper.zypp_config = create_zypp_config(zypper_base, options[:arch])
+        zypper.zypp_config = create_zypp_config(options[:arch])
       end
 
       block.call(zypper)
     ensure
-      clean_up(zypper_base)
+      clean_up
+    end
+
+    def nfs_repos?
+      Dir.glob(File.join(@zypp_base, "etc/zypp/repos.d", "*")).each do |file|
+        File.foreach(file) do |line|
+          return true if line =~ /^baseurl=nfs:\/\//
+        end
+      end
+      false
     end
 
     private
 
-    def clean_up(base_path)
-      LoggedCheetah.run("sudo", "rm", "-rf", base_path)
+    def clean_up
+      if @zypp_base =~ /^\/tmp\/./
+        cmd = ["rm", "-rf", @zypp_base]
+        cmd = cmd.insert(0, "sudo") if nfs_repos?
+        LoggedCheetah.run(*cmd)
+      else
+        raise("This is supposed to never happen, only tmp directories should be used")
```

Comment:
This message is not very helpful, how about `The zypper base directory is not inside of '/tmp'. Aborting...`.